### PR TITLE
Added Dynamic Allocation configuration to Hadoop/Yarn deploy. (+ Idle Cache timeout)

### PR DESCRIPTION
**Pull request by stvoutsin**
PR includes changes to allow Spark/Yarn to dynamically allocate resources.
Motivation for this was to allow multiple users to be able to use the system concurrently, with each user having their own Spark Context, and have it be able to manage the load so that each user gets positive user experience (i.e. Not jobs stuck at 0% for several minutes).

Changes include configuration to allow Spark/Yarn to clear the cache after 60 seconds of being idle, so that executors don't end up eating up resources and not releasing,

Also notes on testing caching, and a suggestion for a change to the ML Notebook to use Spark SQL Caching, so that the notebook gives consistent results & performance.

### CODE REVIEW

File: notes/stv/20210822-validation-multi-user-01.txt [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    ssh -L '3000:monitor:3000' fedora@128.232.227.178
+
+
+
+
+# -----------------------------------------------------
+# Setup a tunnel to the Yarn UI
+
+    ssh -L '8088:master01:8088' fedora@128.232.227.178
+
```
Comment (Zarquan) [inline notes/stv/20210822-validation-multi-user-01.txt:237]: Not clear which machine this is run on. I'm guessing this is a direct ssh connect from laptop to the zeppelin node.

**Review by Zarquan — APPROVED** [review]
Looks good. The experiment notes themselves are very detailed.
Might be useful to add a separate note that summarises the results of the various experiments. Having been away from the project for a couple of weeks I loose track of whether 4min is a good or bad result to get for the a particular notebook cell. Perhaps a table comparing the original, with pyspark cache clear : with SQL cache clear : with two users sharing .. etc.
This is the first time we have done this kind of side by side comparisons so we need to learn how to display the results so they can be understood without needing to know the details.

File: notes/stv/20210822-validation-multi-user-01.txt (hunk 1 above)
Comment (stvoutsin) [reply notes/stv/20210822-validation-multi-user-01.txt:237]: Yes it was run on local laptop, will be clearer in future notes

**Comment by Zarquan** [comment]
No problem, I only noticed it because I'd been working on something 
similar, experimenting with ssh tunnels and proxies.

-- Dave

**Pull request merged**